cmd/sink/routers/routers.go

```diff
@@ -90,25 +93,31 @@ func (c *Controller) writeResource(ctx context.Context, obj *unstructured.Unstru
 			"id", event.ID(),
 			"err", err,
 		)
-	} else {
-		slog.Info(
-			"Successfully wrote object from cloudevent to the database",
-			"objectID", obj.GetUID(),
-			"id", event.ID(),
-		)
+		return err
 	}
+	slog.Info(
+		"Successfully wrote object from cloudevent to the database",
+		"objectID", obj.GetUID(),
+		"id", event.ID(),
+	)
 	// only write logs for k8s resources likes pods that have them and UrlBuilder is configured
 	if c.LogUrlBuilder != nil && strings.ToLower(obj.GetKind()) == "pod" {
-		c.writeLogs(ctx, obj)
+		err = c.writeLogs(ctx, obj)
 	}
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
Why not just:
```
return err
```

Reply:
not sure why I did that. I removed the if at the end of `writeResource` and replaced it with `return err`